tools/clang/lib/AST/ASTContextHLSL.cpp

```diff
@@ -1161,6 +1163,31 @@ CXXRecordDecl *hlsl::DeclareRayQueryType(ASTContext &context) {
   return typeDeclBuilder.getRecordDecl();
 }
 
+CXXRecordDecl *hlsl::DeclareHitObjectType(ASTContext &Context) {
+  // HitObject { ... }
+  BuiltinTypeDeclBuilder TypeDeclBuilder(Context.getTranslationUnitDecl(),
```

Comment:
Since we are not providing a Vulkan implementation of this, we should ensure that all the SER API additions are added to the `dx` namespace.

Reply:
This is in line with all other type declarations living in the hlsl namespace (and there is no dx namespace). The spec makes HitObject part of HLSL, not of the DX backend. Please elaborate what you mean here.

Reply:
Thank you for pointing out the issue with the spec. I've filed this issue tracking updating the spec: https://github.com/microsoft/hlsl-specs/issues/395

The original feedback stands.

Reply:
> With the NodeRecord types we demonstrated a new solution, where we add an unspellable attribute to the type that identifies it as a specific object. We should do something like that here as well.
> 
> If you take a look at how GetNodeIOType is implemented that will give you a model for how this should be done.

HitObject is now modelled after the NodeRecord types as you suggested.

Reply:
> [b028aaf](https://github.com/microsoft/DirectXShaderCompiler/commit/b028aaf6b849d285f307dd7077e68e8bc46a4a56) moves the SER implementation to the `dx` namespace. @llvm-beanz

This change does the wrong thing. You've moved the implementation to a namespace which is not what was requested. The `HitObject` type should be emitted into the AST under the `dx` namespace.

Reply:
[cefafa6](https://github.com/microsoft/DirectXShaderCompiler/pull/7097/commits/cefafa6a7d78fcff5488bac60cc848266c98e265)  moves the SER HitObject type and builtins to the `dx` namespace